project/dexter-monitor/public/ctrl/weeklyGroupCtrl.js

```diff
@@ -91,31 +89,31 @@ monitorApp.controller("DefectByGroupCtrl", function($scope, $http, $log, $q, Def
 
         return $q.all(promises).then(() => {
             for(let i=minYear ; i<=maxYear ; i++) {
-                defect.years.push(i);
+                $scope.years.push(i);
             }
             return DefectService.getMaxWeek(maxYear)
                 .then((week) => {
-                    defect.maxWeekOfCurYear = week;
+                    $scope.maxWeekOfCurYear = week;
                 });
         });
     }
 
-    defect.yearChanged = function(year) {
-        defect.curYear = year;
-        DefectService.getMaxWeek(defect.curYear)
+    $scope.yearChanged = function(year) {
```

Comment:
Please, change the method name: yearChange to setCurrentYearAndReloadData or something to explaint what the method does.